types/index.d.ts

```diff
@@ -4,7 +4,16 @@ import * as Unist from 'unist'
 import * as vfileMessage from 'vfile-message'
 
 declare namespace vfile {
-  type VFileContents = string | Buffer
+  /**
+   * Type must have the same base as Node.js Buffer
+   */
+  type BufferLike = Uint8Array
```

Comment:
in the official types, Buffer extends `Uint8Array` https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f25874237274475768cec58ef1458b1178437dbf/types/node/ts3.2/globals.d.ts#L10-L19

`Uint8Array` is a native ES5 type.

Reply:
I think this is the real official type for Buffer: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/v12/globals.d.ts#L252

I like the idea of using Uint8Array to represent Buffers without requiring node types, but wouldn’t it be simpler to just define this then?

```ts
type VFileContents = string | Uint8Array
```

I would understand though if this is just for documentation purposes.

Reply:
> I think this is the real official type for Buffer: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/v12/globals.d.ts#L252

That may be, for some reason vscode's jump to definition jumps to the ts3.2 version, it seems like the node typing may be some advanced type layering based off the typescript version.

Either way, `Uint8Array` should offer good compatability all the way back to Node 4 https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f25874237274475768cec58ef1458b1178437dbf/types/node/v4/index.d.ts#L567

> wouldn’t it be simpler to just define this then?

It may be

> I would understand though if this is just for documentation purposes.

I was going for documentation, but to your point, the documentation doesn't need to be a type alias, it could be a tsdoc comment.

Reply:
The other possible use would be a utility type for reuse elsewhere in the ecosystem.
But searching the existing typings, it appears this is the only place `Buffer` is referenced directly as a type.

Reply:
Updated in 14c3c95